src/core/lib/event_engine/posix_engine/timer_manager.h

```diff
@@ -73,12 +73,20 @@ class TimerManager final {
     TimerManager* const timer_manager_;
   };
 
+  enum class State {
+    kRunning,   // processing timers
```

Comment:
You're never actually using the other states, they are treated the same (except for logging). All checks are whether `state_ == State::kRunning`.

Reply:
Restart needs to know if the manager was shutdown (so does not need to run) or suspended and needs to be resumed. This is one of the main fixes.

Reply:
I believe you can use a single `bool shutdown_` bit for that.